src/fs_utils.rs

```diff
@@ -47,6 +49,26 @@ pub fn write_file(path: &PathBuf, lines: Vec<LineEntry>) -> io::Result<()> {
     Ok(())
 }
 
+pub fn backup_file(path: &PathBuf) -> Result<PathBuf, Box<dyn Error>> {
+    let timestamp = SystemTime::now()
+        .duration_since(SystemTime::UNIX_EPOCH)?
+        .as_secs();
+
+    let mut new_filepath = PathBuf::new();
+    if let Some(orig_path) = path.to_str() {
+        new_filepath = PathBuf::from(format!(
+            "{orig_path}_{timestamp}",
+            orig_path = orig_path,
+            timestamp = timestamp
+        ));
+    }
+
+    match copy(path, &new_filepath) {
+        Ok(_) => Ok(new_filepath),
+        Err(e) => Err(Box::new(e)),
+    }
+}
```

Comment:
```suggestion
pub fn backup_file(fe: &FileEntry) -> Result<PathBuf, Box<dyn Error>> {
    let timestamp = SystemTime::now()
        .duration_since(SystemTime::UNIX_EPOCH)?
        .as_secs();

    let mut new_path = fe.path.to_owned();
    new_path.set_file_name(format!("{}_{}", &fe.file_name, timestamp));

    match copy(&fe.path, &new_path) {
        Ok(_) => Ok(new_path),
        Err(e) => Err(Box::new(e)),
    }
}
```

I think you can get the file name from the `FileEntry` object and then add a timestamp to the new file path.
What do you think about this way?

Reply:
I think that is much clearer and made that change. Thank you.